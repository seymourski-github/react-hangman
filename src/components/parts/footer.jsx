import React from 'react';
import PropTypes from 'prop-types';
import { Button, Container, Label } from '../../components';

export const Footer = props => {

  const {
    action, id, labels, name, result, handleAction
  } = props;

  const counter = labels.counter(props);

  return (
    <Container flex={true} id={id} >
      <Label text={counter} visible={!result} />
      <Button action={action} name={name} handleClick={handleAction} />
    </Container>
  );
};

Footer.propTypes = {
  action: PropTypes.string,
  labels: PropTypes.object,
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  handleAction: PropTypes.func
};

Footer.defaultProps = {
  action: null,
  labels: {},
  handleAction: () => console.log('handleAction button fail')
};
