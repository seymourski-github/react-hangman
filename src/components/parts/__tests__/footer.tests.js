import React from 'react';
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { Footer } from '../footer';

const props = {
  id: 'my-footer-panel',
  name: 'restart',
  labels: {
    counter: props => 'my counter'
  }
};

describe('<Guesses /> Component', function() {

  const component = renderer.create(<Footer {...props} />);
  let wrapper = shallow(<Footer {...props} />);

  it("renders and matches our snapshot", () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
