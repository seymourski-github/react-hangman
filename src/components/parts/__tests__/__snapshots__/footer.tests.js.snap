// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Guesses /> Component renders and matches our snapshot 1`] = `
<div
  className="container row"
  id="my-footer-panel"
  style={undefined}
>
  <label
    className="label"
    disabled={false}
    hidden={false}
    htmlFor={undefined}
    style={
      Object {
        "alignSelf": "",
        "flex": undefined,
        "visibility": "visible",
      }
    }
  >
    my counter
  </label>
  <button
    className="button"
    disabled={false}
    name="restart"
    onClick={[Function]}
    style={null}
    type="button"
    value=""
  >
    restart
  </button>
</div>
`;
